src/atomate2/common/builders/magnetism.py

```diff
@@ -0,0 +1,248 @@
+            for r_task in relax_tasks:
+                if r_task["uuid"] == task["metadata"]["parent_uuid"]:
+                    relax_output = self._build_relax_output(
+                        r_task["output"],
+                        uuid=r_task["uuid"],
+                    )
+                    break
+            output = self._build_static_output(
+                task["output"],
+                uuid=task["uuid"],
+            )
+            if relax_output is not None:
+                energy_diff = output.energy_per_atom - relax_output.energy_per_atom
+            output.relax_output = relax_output
+            output.energy_diff_relax_static = energy_diff
+            outputs.append(output)
+
+        return jsanitize(
+            MagneticOrderingsDocument.from_outputs(
+                outputs, parent_structure=parent_structure
+            ).dict(),
+            allow_bson=True,
+        )
+
+    def update_targets(self, items: list[MagneticOrderingsDocument]):
+        """
+        Insert new magnetic orderings into the magnetic orderings Store.
+
+        Parameters
+        ----------
+        items : list of .MagneticOrderingsDocument
+            A list of magnetic ordering documents to add to the database.
+        """
+        self.logger.info(f"Updating {len(items)} magnetic orderings documents")
+        self.magnetic_orderings.update(items, key="ground_state_uuid")
+
+    def _build_relax_output(self, relax_task, uuid=None) -> MagneticOrderingRelaxation:
+        """Wrapper for MagneticOrderingRelaxation.from_task_document."""
+        raise NotImplementedError
+
+    def _build_static_output(self, static_task, uuid=None) -> MagneticOrderingOutput:
+        """Wrapper for MagneticOrderingOutput.from_task_document."""
+        raise NotImplementedError
+
+    @property
+    def _dft_code_query(self):
+        """
+        Return a query criterion for querying the tasks collection for DFT code-specific
+        tasks. This is to be implemented for the DFT code of choice.
+
+        This prevents mixing of tasks from different DFT codes in the same builder.
+        """
+        raise NotImplementedError
+
+
+def _group_orderings(tasks: list[dict], tol: float) -> list[list[dict]]:
+    """
+    Group deformation tasks by their parent structure.
+
+    Parameters
+    ----------
+    tasks : list of dict
+        A list of deformation tasks.
+    tol : float
+        Numerical tolerance for structure equivalence.
+
+    Returns
+    -------
+    list of list of dict
+        The tasks grouped by their parent (undeformed structure).
+    """
+    grouped_tasks = [[tasks[0]]]
+
+    for task in tasks[1:]:
+        parent_structure = task["metadata"]["parent_structure"]
+
+        match = False
+        for group in grouped_tasks:
+            group_parent_structure = task["metadata"]["parent_structure"]
+
+            # parent structure should really be exactly identical (from same workflow)
```

Comment:
I like this function. I wonder if we might benefit from the same code but in an `exact_match()` utility function somewhere instead, it seems like it might be useful more generally.